pkg/block/namespace.go

```diff
@@ -146,6 +152,20 @@ func ResolveNamespace(defaultNamespace, key string) (QualifiedKey, error) {
 	}, nil
 }
 
+func parseKey(key string) (parsedKey *url.URL, relative bool, err error) {
+	parsedKey, err = url.ParseRequestURI(key)
+	if err == nil {
+		return parsedKey, false, nil
+	}
+	if strings.Contains(key, "://") {
+		// No better way to understand if the key failed to parse since it's relative
+		// or due to bad formatting. See https://github.com/treeverse/lakeFS/issues/1646
```

Comment:
Not sure I understand.  `s3://a/path/with/a/prefix/ending/in/://and/an/empty/component` is a perfectly valid path. We should be able to do better, these are URLs after all, and have a unique parse.